dart/biomechanics/SubjectOnDisk.cpp

```diff
@@ -2147,12 +2148,29 @@ void SubjectOnDiskTrialPass::computeValuesFromForcePlates(
 
   std::vector<int> footIndices;
   std::vector<dynamics::BodyNode*> footBodies;
+  bool foundNonexistentFootBody = false;
   for (std::string footName : footBodyNames)
   {
     dynamics::BodyNode* footBody = skel->getBodyNode(footName);
+
+    if (!footBody) {
+      foundNonexistentFootBody = true;
+      break;
+    }
+
     footBodies.push_back(footBody);
     footIndices.push_back(footBody->getIndexInSkeleton());
   }
+
+  if (foundNonexistentFootBody) {
+    std::cout << "WARNING: One of the foot bodies specified does not exist in "
+                 "the skeleton. Skipping dynamics calculations..."
+              << std::endl;
+    computeKinematicValues(skel, timestep, poses, rootHistoryLen,
```

Comment:
Should we maybe move this line prior to the if statement, so it always gets called, and then just exit early in here? That way we can delete the code down below that computes the same values, and avoid duplication.

Reply:
Good idea, will do!

Reply:
Done.